src/thread_manager.ts

```diff
@@ -33,14 +32,17 @@ export class ThreadManager<SCG extends ExtendableGenerics = DefaultGenerics> {
   public readonly state: StateStore<ThreadManagerState<SCG>>;
   private client: StreamChat<SCG>;
   private unsubscribeFunctions: Set<() => void> = new Set();
+  private threadsByIdGetterCache: {
+    threads: ThreadManagerState<SCG>['threads'];
+    threadsById: { [key: string]: Thread<SCG> };
```

Comment:
I would advise `{ [key: string]: Thread<SCG> | undefined }` (or just `Record<string, Thread<SCG> | undefined>`), because without `noUncheckedIndexAccess` this is just more true :)

Reply:
Good point, fixing right away.